lib/queue.js

```diff
@@ -545,9 +544,8 @@ class Queue extends Emitter {
   }
 
   _finishJob(err, data, job) {
```

Comment:
Refactored this method and moved some code to `Job#computeDelay` so we can compute the final `status` early.